presto-docs/src/main/sphinx/admin/function-namespace-managers.rst

```diff
@@ -0,0 +1,76 @@
+===========================
+Function Namespace Managers
+===========================
+
+.. warning::
+
+    This is an experimental feature being actively developed. The way
+    Function Namespace Managers are configured might be changed.
+
+Function namespace managers support storing and retrieving user-defined
+functions with a persistent storage, allowing users to create, alter,
+and delete functions, as well as to reference them within queries.
+
+Function Namespace
+------------------
+
+A function namespace is a catalog-schema name (i.e. ``example.test``)
+in which each function can be uniquely identified by its signature,
+that is, its name and parameter type list.
+
+The function namespace, such as ``example.test``, is not a full-fledged
```

Comment:
"A function namespace"? What's the behavior if I use a real catalog as function catalog? I think it will still work right? So potentially "example.text" can be a real catalog.schema as well.

Reply:
I think we should disallow registering function namespace with a catalog that has already been registered with a connector. This will cause lots of confusion as the sets of schema will be different from the sets of function namespaces within that catalog, and makes support `CREATE SCHEMA`, `DROP SCHEMA`, and `SHOW SCHEMA` on function namespace managed catalogs problematic or even infeasible.

Reply:
If we have connector functions one day, we can reverse that restriction and properly support that use case.

Reply:
But we actually don't disallow it in code right? So call this out specifically. Something like using existing connector catalog is not supported and the behavior is not defined or tested.

Reply:
```
Each instance of function namespace manager binds to a catalog name,
as it manages all functions within that catalog. Using the catalog
name of an existing connector is discouraged, as the behavior is not
defined nor tested, and will be disallowed in the future.

Currently, those catalog names do not corresponds to real catalogs
as they cannot be specified as the catalog in session, nor do they
support :doc:`CREATE SCHEMA`, :doc:`ALTER SCHEMA`, :doc:`DROP SCHEMA`,
or :doc:`SHOW SCHEMAS`. Instead, namespaces can be added using the
methods described below.
```